src/Microsoft.ML.OnnxTransform/OnnxTransform.cs

```diff
@@ -0,0 +1,445 @@
+using Microsoft.ML.Runtime.Model;
+using Microsoft.ML.Scoring;
+using Microsoft.ML.OnnxScoring;
+using OnnxShape = System.Collections.Generic.List<long>;
+using Microsoft.ML.Core.Data;
+using Microsoft.ML.Data.StaticPipe.Runtime;
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform),
+    typeof(OnnxTransform.Arguments), typeof(SignatureDataTransform), OnnxTransform.UserName, OnnxTransform.ShortName)]
+
+[assembly: LoadableClass(OnnxTransform.Summary, typeof(IDataTransform), typeof(OnnxTransform),
+    null, typeof(SignatureLoadDataTransform), OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(OnnxTransform), null, typeof(SignatureLoadModel),
+    OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: LoadableClass(typeof(IRowMapper), typeof(OnnxTransform), null, typeof(SignatureLoadRowMapper),
+    OnnxTransform.UserName, OnnxTransform.LoaderSignature)]
+
+[assembly: EntryPointModule(typeof(OnnxTransform))]
+
+namespace Microsoft.ML.OnnxScoring
+{
+    public sealed class OnnxTransform : ITransformer, ICanSaveModel
+    {
+        public sealed class Arguments : TransformInputBase
+        {
+
+            [Argument(ArgumentType.Required, HelpText = "Path to the onnx model file.", ShortName = "model", SortOrder = 0)]
+            public string ModelFile;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "TBD", SortOrder = 1)]
+            public string InputColumn;
+
+            [Argument(ArgumentType.Multiple | ArgumentType.Required, HelpText = "TBD", SortOrder = 2)]
+            public string OutputColumn;
+        }
+
+        private readonly IHost _host;
+        private readonly Arguments _args;
+        private readonly OnnxModel _model;
+        private const string RegistrationName = "OnnxTransform";
+
+        internal const string Summary = "Transforms the data using the Onnx model.";
+        internal const string UserName = "OnnxTransform";
+        internal const string ShortName = "Onnx";
+        internal const string LoaderSignature = "OnnxTransform";
+
+        public readonly string[] Inputs;
+        public readonly string[] Outputs;
+
+        private static VersionInfo GetVersionInfo()
+        {
+            return new VersionInfo(
+                modelSignature: "ONNXSCOR",
+                verWrittenCur: 0x00010001, // Initial
+                verReadableCur: 0x00010001,
+                verWeCanReadBack: 0x00010001,
+                loaderSignature: LoaderSignature);
+        }
+
+        // Factory method for SignatureDataTransform
+        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        {
+            return new OnnxTransform(env, args).MakeDataTransform(input);
+        }
+
+        // Factory method for SignatureLoadDataTransform
+        public static IDataTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
```

Comment:
>public [](start = 8, length = 6)

can make private

Reply:
same for all factory methods

---
In reply to: [219613072](https://github.com/dotnet/machinelearning/pull/942#discussion_r219613072) [](ancestors = 219613072)

Reply:
Some, yes, probably one which returns IDataTransform. All other methods, just for our dependency framework, and can be private. 

---
In reply to: [220260318](https://github.com/dotnet/machinelearning/pull/942#discussion_r220260318) [](ancestors = 220260318)

Reply:
Made them private

---
In reply to: [219613072](https://github.com/dotnet/machinelearning/pull/942#discussion_r219613072) [](ancestors = 219613072)

Reply:
Ok, thanks Ivan. We've made them private for now -- all unit tests are passing so far.

---
In reply to: [220309543](https://github.com/dotnet/machinelearning/pull/942#discussion_r220309543) [](ancestors = 220309543,220260318)